File: DesktopApp/Praktikumsverwaltung_DesktopApp/Praktikumsverwaltung_DesktopApp/pkgData/Entry.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Praktikumsverwaltung_DesktopApp.pkgData
{    
    class Entry
    {
        public string Id { get; set; }
        public DateTime StartDate { get; set; }           // DateTime as String because otherwise deserialization causes problems
        public DateTime EndDate { get; set; }
        public double Salary { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public bool AllowedTeacher { get; set; }
        public bool AllowedAV { get; set; }
        public bool SeenByAdmin { get; set; }
        public string AdminNote { get; set; }
        public string IdPupil { get; set; }
        public string IdCompany { get; set; }
        public string IdClass { get; set; }

        public Entry(string id, DateTime startDate, DateTime endDate, string title, string description, double salary, bool allowedTeacher, bool allowedAV, bool seenByAdmin, string adminNote, string idPupil, string idClass, string idCompany)
        {
            Id = id;
            StartDate = startDate;
            EndDate = endDate;
            Title = title;
            Description = description;
            Salary = salary;
            AllowedTeacher = allowedTeacher;
            AllowedAV = allowedAV;
            SeenByAdmin = seenByAdmin;
            AdminNote = adminNote;
            IdPupil = idPupil;
            IdCompany = idCompany;
            IdClass = idClass;
        }
    }
}

File: PraktikumsverwaltungWebService/src/java/pkgData/Entry.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pkgData;

import java.time.LocalDate;
import javax.xml.bind.annotation.XmlRootElement;
import org.bson.types.ObjectId;

/**
 *
 * @author schueler
 */
@XmlRootElement
public class Entry {
    private String id;
    private LocalDate startDate;
    private LocalDate endDate;
    private double salary;
    private String title;
    private String description;
    private boolean allowedTeacher;
    private boolean allowedAV;
    private boolean seenByAdmin;
    private String idPupil;
    private String idCompany;
    private String idClass;

    public Entry() {
    }

    public Entry(String id, LocalDate startDate, LocalDate endDate, double salary, String title, String description, boolean allowedTeacher, boolean allowedAV, boolean seenByAdmin, String idPupil, String idCompany, String idClass) {
        this.id = id;
        this.startDate = startDate;
        this.endDate = endDate;
        this.salary = salary;
        this.title = title;
        this.description = description;
        this.allowedTeacher = allowedTeacher;
        this.allowedAV = allowedAV;
        this.seenByAdmin = seenByAdmin;
        this.idPupil = idPupil;
        this.idCompany = idCompany;
        this.idClass = idClass;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public LocalDate getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDate startDate) {
        this.startDate = startDate;
    }

    public LocalDate getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDate endDate) {
        this.endDate = endDate;
    }

    public double getSalary() {
        return salary;
    }

    public void setSalary(double salary) {
        this.salary = salary;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public boolean isAllowedTeacher() {
        return allowedTeacher;
    }

    public void setAllowedTeacher(boolean allowedTeacher) {
        this.allowedTeacher = allowedTeacher;
    }

    public boolean isAllowedAV() {
        return allowedAV;
    }

    public void setAllowedAV(boolean allowedAV) {
        this.allowedAV = allowedAV;
    }

    public boolean isSeenByAdmin() {
        return seenByAdmin;
    }

    public void setSeenByAdmin(boolean seenByAdmin) {
        this.seenByAdmin = seenByAdmin;
    }

    public String getIdPupil() {
        return idPupil;
    }

    public void setIdPupil(String idPupil) {
        this.idPupil = idPupil;
    }

    public String getIdCompany() {
        return idCompany;
    }

    public void setIdCompany(String idCompany) {
        this.idCompany = idCompany;
    }

    public String getIdClass() {
        return idClass;
    }

    public void setIdClass(String idClass) {
        this.idClass = idClass;
    }
    
    
}
